exporters/prometheus/exporter.go

```diff
@@ -344,3 +375,27 @@ func sanitizeName(n string) string {
 
 	return b.String()
 }
+
+func validateMetrics(name, description string, metricType *dto.MetricType, mfs map[string]*dto.MetricFamily) (drop bool, help string) {
+	emf, exist := mfs[name]
+	if !exist {
+		mfs[name] = &dto.MetricFamily{
+			Name: proto.String(name),
+			Help: proto.String(description),
+			Type: metricType,
+		}
+		return false, ""
+	}
+	if emf.GetHelp() != description {
+		if emf.GetType() == dto.MetricType_HISTOGRAM || *metricType == dto.MetricType_HISTOGRAM {
```

Comment:
Why don't we compare descriptions for histograms?

Reply:
Because I want to skip the case that create counter and histogram with same name, as following
```
meterA.SyncInt64().Counter("foo",...)

meterA.SyncInt64().Histogram("foo"...)
```
although the two instrument are with same name, but the histogram will finally create metric name with suffix as `foo_bytes_sum`, `foo_bytes_count` and `foo_bytes_bucket` which are different name with the counter's metric name `foo_bytes_total`. In this case, prometheus won't throw error.

Do you think we should better disallow this usage pattern? @dashpole

Reply:
I think the problem is that what if you have two histograms with different descriptions?  That is probably grounds for an additional test case.

Reply:
It will still throw an error because there are two identical sets of comments.  I.e.
```
# TYPE foo histogram
# TYPE foo counter
```
Isn't allowed

Reply:
That is a good point, I will check about this. @MadVikingGod @dashpole

Reply:
I fixed this problem, and add more test for conflict help to cover 3 cases for `Counter` `UpDownCounter`, `Histogram`.
@MadVikingGod @dashpole